aws-s3/getmessage_metric.py

```diff
@@ -270,35 +271,36 @@ def yield_message(message):
         yield metric
 
 
-def process_get_data(logger, cli_config_vars, method, url):
-    logger.info('start to request data')
+def process_get_data(logger, cli_config_vars, bucket, object_key):
+    logger.info('start to process data from file: {}'.format(object_key))
     global messages
-    req = requests.get
-    if method.upper() == 'POST':
-        req = requests.post
-    response = req(url)
-    if response.status_code == http.client.OK:
-        message = response.text.splitlines()
-        message = [json.loads(metric_data) for metric_data in message]
-        threadLock.acquire()
-        messages.extend(message)
-        threadLock.release()
-    else:
-        logger.error('{} failed to get data'.format(url))
-
-    logger.info('Closed data process......')
-
-
-class myThread(threading.Thread):
-    def __init__(self, logger, cli_config_vars, method, url):
+
+    buf = io.BytesIO()
+    logger.info('downloading file: {}'.format(object_key))
+
+    bucket.download_fileobj(object_key, buf)
+    content = buf.getvalue()
+    logger.info('download {} bytes from file: {}'.format(len(content), object_key))
```

Comment:
If the files are very large, this might be an issue. I don't think we need to change this now, but something for us to keep in mind

Reply:
I have test with several 50M+ files, the speed is good.